File: frontend/src/types/child.ts
export interface Child {
  id: number;
  user_id: number;
  name: string;
  date_of_birth: string;
  gender?: string;
  notes?: string;
  created_at: string;
  updated_at: string;
}

export interface ChildCreate {
  name: string;
  date_of_birth: string;
  gender?: string;
  notes?: string;
}

export interface ChildUpdate {
  name?: string;
  date_of_birth?: string;
  gender?: string;
  notes?: string;
}

export function calculateAge(dateOfBirth: string): number {
  const today = new Date();
  const birthDate = new Date(dateOfBirth);
  let age = today.getFullYear() - birthDate.getFullYear();
  const monthDiff = today.getMonth() - birthDate.getMonth();

  if (monthDiff < 0 || (monthDiff === 0 && today.getDate() < birthDate.getDate())) {
    age--;
  }

  return age;
}

File: app/schemas/child.py
"""Child profile request/response schemas."""
from datetime import date
from typing import Optional
from pydantic import BaseModel, Field


class ChildCreate(BaseModel):
    """Create child profile request."""
    name: str = Field(min_length=1, max_length=255)
    date_of_birth: date
    gender: Optional[str] = Field(None, max_length=50)
    notes: Optional[str] = None


class ChildUpdate(BaseModel):
    """Update child profile request."""
    name: Optional[str] = Field(None, min_length=1, max_length=255)
    date_of_birth: Optional[date] = None
    gender: Optional[str] = Field(None, max_length=50)
    notes: Optional[str] = None


class ChildResponse(BaseModel):
    """Child profile response."""
    id: int
    name: str
    date_of_birth: date
    gender: Optional[str]
    notes: Optional[str]
    age_years: int
    created_at: str
    updated_at: str

    class Config:
        from_attributes = True
